worlds/checksmate/Options.py

```diff
@@ -0,0 +1,415 @@
+    """
+    display_name = "Queen Piece Limit"
+    range_start = 0
+    range_end = 7
+    default = 0
+    special_range_cutoff = 1
+
+
+class ItemLimitByType(ItemDict):
+    """
+    NOT IMPLEMENTED
+
+    For Minor Pieces, Major Pieces, and Major To Queens, limits the amount of any given type of equivalent piece you
+    might play with. For example, if you limit Progressive Queen To Major to 1, you will never see 2 Amazons, although
+    you might see 1 Amazon and 1 Colonel.
+
+    For Progressive Pocket, limits the amount which might be distributed to an individual pocket. If this is set to 1,
+    any given Pocket will never hold anything more substantial than a Pawn. If this is set to 3, any given Pocket will
+    never hold a Queen. The default of 4 allows each of the 3 spaces to hold between 0-4 progressive items.
+
+    No effect on other settings.
+
+    NOT IMPLEMENTED
+    """
+    display_name = "Item Limit By Type"
+
+
+class MaximumItems(ItemDict):
+    """
+    NOT IMPLEMENTED
+
+    For Major To Queen, how many Queen-equivalent pieces you might play with. If set to 1, you will never have more than
+    1 piece upgraded to a Queen. (This does nothing when greater than 'Queen Piece Limit by Type'.) You may still
+    promote pawns during a game. If set to 0, this setting is disabled.
+
+    For Progressive Consul, how many Royal pieces (Kings) to place, which must all be captured before one experiences
+    defeat. The player always starts with 1 King, but may find Progressive Consuls if this is set. Progressive Consuls
+    add additional Kings to the player's starting board.
+
+    For Progressive Pocket, the number of Progressive Pocket Pieces the game is allowed to add to the multiworld. Each
+    Progressive Pocket Piece will improve your 1st, 2nd, or 3rd pocket slot up to 4 times, from Nothing to Pawn, to
+    Minor Piece (like a pocket Knight), to Major Piece (like a pocket Rook), to Queen. This setting does not alter
+    filler item distribution. (Even if you have 0 Progressive Pockets, the item pool may contain Progressive Pocket Gems
+    and Progressive Pocket Range.)
+
+    No effect on other settings.
+
+    NOT IMPLEMENTED
+    """
+    display_name = "Maximum Items"
+
+
+class LockedItems(ItemDict):
+    """
+    Guarantees that these progression and filler items will be unlockable.
+
+    Implementation note: Currently forces this many items into the item pool before distribution begins. This behaviour
+    is not guaranteed - a future version may simply validate the pool contains these items.
+    """
+    display_name = "Locked Items"
+
+
+class DeathLink(Toggle):
+    """
+    If on:
+
+    - Whenever you are checkmated or resign (close the game window), everyone who is also on Death Link dies.
+
+    - Whenever you receive a Death Link event, your game window closes. (You cannot undo or review.)
+    """
+    display_name = "Death Link"
+
+
+cm_options: Dict[str, type(Option)] = {
```

Comment:
Will that mean the apworld will require 0.4.4? I am completely fine if that's the case, just want to know what to specify for minimum AP version.

Reply:
Yes, that will mean it will require 0.4.4

Reply:
I've added a note that it will require 0.4.4 on the Releases page for the client. More discussion can be seen in the Chess future game discussion channel on the server.